src/main/java/io/github/nucleuspowered/nucleus/modules/kit/commands/firstjoin/FirstKitRedeemCommand.java

```diff
@@ -26,21 +23,29 @@
 @NoCooldown
 @NoWarmup
 @RegisterCommand(value = "redeem", subcommandOf = FirstKitCommand.class)
-public class FirstKitRedeemCommand extends AbstractCommand<Player> {
-
-    @Inject
-    private KitService gds;
+public class FirstKitRedeemCommand extends AbstractCommand.SimpleTargetOtherPlayer {
+    private final String playerKey = "subject";
 
+    @Inject private KitService gds;
     @Inject private KitConfigAdapter kca;
 
     @Override
-    public CommandResult executeCommand(Player src, CommandContext args) throws Exception {
-
-        Tristate tristate = Util.addToStandardInventory(src, gds.getFirstKit(), false, kca.getNodeOrDefault().isProcessTokens());
-
-        src.sendMessage(plugin.getMessageProvider().getTextMessageWithFormat("command.firstkit.redeem.success"));
-        if (tristate != Tristate.TRUE) {
-            src.sendMessage(plugin.getMessageProvider().getTextMessageWithFormat("command.firstkit.redeem.reject"));
+    protected CommandResult executeWithPlayer(CommandSource src, Player pl, CommandContext args, boolean isSelf) throws Exception {
+        Player player = this.getUserFromArgs(Player.class, src, playerKey, args);
+
+        Tristate tristate = Util.addToStandardInventory((Player) src, gds.getFirstKit(), false, kca.getNodeOrDefault().isProcessTokens());
+
+        if (isSelf) {
+            src.sendMessage(plugin.getMessageProvider().getTextMessageWithFormat("command.firstkit.redeem.success"));
```

Comment:
This if statement basically has two duplicate branches in. You could just put the `isSelf` check later and put the `pl.sendMessage` in there - try to reduce duplication as much as possible!